File: frontend/types/goal.ts
// Goal types

export interface Goal {
  goalId: number;
  title: string;
  description: string;
  createdAt: string;
  targetDate: string;
  journalMarkdown?: string | null;
}

export interface GoalWithDetails extends Goal {
  specific?: string;
  measurable?: string;
  attainable?: string;
  relevant?: string;
  timely?: string;
  stepsTaken?: string;
  journalMarkdown?: string | null;
}

export interface CreateGoalRequest {
  title: string;
  description: string;
  targetDate: string;
  specific?: string;
  measurable?: string;
  attainable?: string;
  relevant?: string;
  timely?: string;
  stepsTaken?: string;
  journalMarkdown?: string | null;
}

export type UpdateGoalRequest = Partial<CreateGoalRequest>;

File: backend/src/main/java/com/nagai/backend/goals/Goal.java
package com.nagai.backend.goals;
import java.time.LocalDateTime;

import org.hibernate.annotations.CreationTimestamp;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Data;

@Entity
@Data
@Table(name="goals")
public class Goal {

    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    @Column(name="goal_id")
    private Long goalId;

    @Column(name="user_id")
    private Long userId;

    private String title;
    private String description;
    @Column(name="target_date")
    private String targetDate;
    private String specific;
    private String measurable;
    private String attainable;
    private String relevant;
    private String timely;

    @Column(name="created_at")
    @CreationTimestamp
    private LocalDateTime createdAt;
}
